### PR TITLE
feat(variable-dropdown): apply new use-context-menu-controller & update mirinae version

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [x] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

SSIA

- Apply new `use-context-menu-controller`
- Update mirinae version

### CODE REVIEW

File: src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue [hunk 1]
```diff
@@ -118,18 +132,34 @@ const handleOpenOverlay = () => {
...
+const handleUpdateSearchText = debounce((text: string) => {
+    searchText.value = text;
+    reloadMenu();
+}, 200);
+
+watch(visibleMenu, (_visibleMenu) => {
+    if (_visibleMenu) {
+        initiateMenu();
+    }
+}, { immediate: true });
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue:161]: nitpick: I think `visibleMenu.value` should be watched!

```suggestion
watch(visibleMenu.value, (_visibleMenu) => {
    if (_visibleMenu) {
        initiateMenu();
    }
}, { immediate: true });
```

File: src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue [hunk 2]
```diff
@@ -172,11 +170,22 @@ const changeVariables = (changedSelected: MenuItem[]) => {
     dashboardDetailState.variables = variables;
 };
 
+const handleUpdateSearchText = debounce((text: string) => {
+    searchText.value = text;
+    reloadMenu();
+}, 200);
+
+watch([options, visibleMenu], ([, _visibleMenu]) => {
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue:178]: nitpick:

```suggestion
watch([options.value, visibleMenu.value], ([, _visibleMenu]) => {
```

or

```suggestion
watch([state.options, visibleMenu.value], ([, _visibleMenu]) => {
```

**Review by yuda110 — APPROVED** [review]
LGTM! I left some comments

File: src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue [hunk 3]
```diff
@@ -79,26 +85,26 @@ const state = reactive({
...
-    originMenu: variableList,
-    selected,
+    targetRef: toRef(state, 'targetRef'),
+    contextMenuRef: toRef(state, 'contextMenuRef'),
+    useMenuFiltering: true,
+    useReorderBySelection: true,
+    searchText: toRef(state, 'searchText'),
+    selected: toRef(state, 'selected'),
+    menu: toRef(state, 'variableList'),
+    pageSize: 5,
```
Comment (WANZARGEN) [inline src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue:107]: question: pageSize 5 is too small, I think 😅... Is this the confirmation from PM?

File: src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue (hunk 1 above)
Comment (WANZARGEN) [reply src/services/dashboards/dashboard-customize/modules/VariableMoreButtonDropdown.vue:153]: No this is correct..

**Review by WANZARGEN — APPROVED** [review]
Thanx for your hard work :)
I left a very tiny tiny tiny comment :)

**Pull request merged**